lib/node/runner/node_runner.go

```diff
@@ -548,6 +552,8 @@ func (nr *NodeRunner) handleBallotMessage(msg interface{}) (err error) {
 			nr.log.Debug("failed to handle ballot", "error", err, "state", baseChecker.Ballot.State())
 			return
 		}
+
+		api.TriggerEvent(nr.storage, checker.StoredProposedTransactions)
```

Comment:
This part triggers the confirmed transactions, right? If right, this is not proper place to run. How about run it as goroutine in the last `runner.saveBlock()`?

Reply:
Ah.. it was my mistake.
as-is code trigger event for every ballot ^^;;;
It should be in the function `func FinishedBallotStore(c common.Checker, args ...interface{}) error {`

Thx!